sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/RealtimeNotificationConfigClient.java

```diff
@@ -0,0 +1,123 @@
+package com.azure.android.communication.chat.implementation.notifications.signaling;
+
+import com.azure.android.core.http.HttpCallback;
+import com.azure.android.core.http.HttpClient;
+import com.azure.android.core.http.HttpMethod;
+import com.azure.android.core.http.HttpRequest;
+import com.azure.android.core.http.HttpResponse;
+import com.azure.android.core.logging.ClientLogger;
+import com.azure.android.core.util.CancellationToken;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class RealtimeNotificationConfigClient {
+    private final HttpClient httpClient;
+
+    private final ClientLogger logger = new ClientLogger(RealtimeNotificationConfigClient.class);
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+
+    RealtimeNotificationConfigClient() {
+        this.httpClient = HttpClient.createDefault();
+    }
+
+    /**
+     * Fetches the RealTimeNotificationConfiguration from the server.
+     *
+     * @param token             The Bearer token for authorization.
+     * @param endpoint          The base endpoint URL.
+     * @param configApiVersion  The API version parameter.
+     */
+    public RealTimeNotificationConfig getTrouterSettings(String token, String endpoint, String configApiVersion) {
+        /// Construct the URL
+        String urlString = endpoint + "/chat/config/realTimeNotifications?api-version=" + configApiVersion;
+
+        // Build the HttpRequest
+        HttpRequest request = new HttpRequest(HttpMethod.GET, urlString);
+        request
+            .setHeader(AUTHORIZATION_HEADER, "Bearer " + token)
+            .setHeader("Accept", "application/json");
+
+        // Initialize CountDownLatch and error holder
+        CountDownLatch latch = new CountDownLatch(1);
+        final Throwable[] requestError = { null };
+        final RealTimeNotificationConfig[] configResult = {null};
+
+        // Send the request asynchronously
+        this.httpClient.send(request, CancellationToken.NONE, new HttpCallback() {
```

Comment:
I can't remember the HttpClient interface. Since this method is sync, do we really need to send HTTP requests async? Does HttpClient offer a sync version of the send method?

Reply:
This httpClient is imported from com.azure.android.core.http.HttpClient. I think Azure team has a requirement on using this instead of Java httpClient, since other classes are using this library as well. The Azure version of HttpClient is designed to make aync call. My understanding is that Android has its own UI. So the client is designed to handle the request in a separate thread instead of the main thread.